CMakeLists.txt

```diff
@@ -22,21 +22,12 @@ option(MAGMA_ENABLE_CUDA     "Enable the CUDA backend"  OFF)
 option(MAGMA_ENABLE_HIP      "Enable the HIP  backend"  OFF)
 
 # check if one backend has been enabled
-if (NOT MAGMA_ENABLE_CUDA AND
-    NOT MAGMA_ENABLE_HIP
-    )
+if(NOT MAGMA_ENABLE_CUDA AND NOT MAGMA_ENABLE_HIP)
   message(STATUS "MAGMA requires one enabled backend!")
   message(STATUS "Building CUDA backend")
   set( MAGMA_ENABLE_CUDA ON )
 endif()
 
-# ----------------------------------------
-# don't regenerate files during make.
-# (I think this means you have to manually re-run CMake if CMakeLists changes.
-# It fixes the huge problems with CMake interrupting Visual Studio.)
-set(CMAKE_SUPPRESS_REGENERATION on)
-
-
```

Comment:
I'm inclined to keep this. Maybe VS + CMake has improved now, but years ago it was essentially impossible to compile MAGMA in VS without setting this flag.

Reply:
Because the minimum version is now 3.18. It should be possible.